### PR TITLE
add vis/uuid/static/:w/:h.png

### LINKED ISSUE #4153: add vis/uuid/static/:w/:h.png
this new endpoint should redirect to the named map url.
so

```
https://rambo-test.cartodb.com/viz/ed64a3de-fd42-11e4-8ae4-0e49835281d6/static/width/height.png
```

redirects to 

```
https://rambo-test.cartodb.com/api/v1/map/static/named/tpl_d64a3de_fd42_11e4_8ae4_0e49835281d6/width/height.png
```

The redirection should have cache headers and should return both http and https versions

Ideally it should redirect to CDN url version of that assets but I'm not sure we have that information in rails process

```
https://cartocdn-ashbu.global.ssl.fastly.net/rambo-test/api/v1/map/static/named/tpl_d64a3de_fd42_11e4_8ae4_0e49835281d6/width/height.png
```

Branch: `4153-static-maps-redirects`

### PR DESCRIPTION
**Pull request by Kartones**
Closes #4153

Note: `normalize_name` change to `template_name` is because old one was not as clear, new one fits better, but is a simple replace without touching logic.

### CODE REVIEW

**Comment by Kartones** [comment]
@juanignaciosl CR please

@javisantana take a look, I've implemented the desired behaviour and alternatives, but just in case I missed something. I haven't pushed any config change because I prefer to leave CDN Details to @CartoDB/systems or to you.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4582/))

File: app/controllers/carto/api/visualizations_controller.rb [hunk 1]
```diff
@@ -96,8 +97,55 @@ def list_watching
         render_jsonp(watcher.list)
       end
 
+      def static_map
+        visualization = VisualizationQueryBuilder.new.with_id(params.fetch('id', nil)).build.first
```
Comment (juanignaciosl) [inline app/controllers/carto/api/visualizations_controller.rb]: Why `params.fetch('id', nil)` and attempting the search? Shouldn't it return a 400 if it has no id?

**Comment by juanignaciosl** [comment]
:+1:

File: app/controllers/carto/api/visualizations_controller.rb (hunk 1 above)
Comment (Kartones) [inline app/controllers/carto/api/visualizations_controller.rb]: Below was doing a `nil` check to indeed fail, but that code is redundant now as there's a `before_filter` -> `load_visualization`, thanks for pointing this out!

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4593/))

**Comment by Kartones** [comment]
@CartoDB/systems I need help testing this at staging. I'm getting consistent 404s on all requests almost 100% because it ends in `.png` and in local works, code and Rails routes are proper ones, etc. 

I'm trying to pinpoint the issue checking logs but with experienced help it would be quicker.

**Comment by Kartones** [comment]
Ok, confirmed the issue is not on the code but server configuration.

If I change Rails route from
```
get '(/user/:user_domain)(/u/:user_domain)/api/v2/viz/:id/static/:width/:height.png'
```
to 
```
get '(/user/:user_domain)(/u/:user_domain)/api/v2/viz/:id/static/:width/:height.test'
```
and request a static map with `/api/v2/viz/6ffe6338-0d39-11e5-a850-040120e0c101/static/300/300.test` it works, while `/api/v2/viz/6ffe6338-0d39-11e5-a850-040120e0c101/static/300/300.png` throws a 404.

Also confirmed that with `.png` Unicorn gets no request.

So I guess I should wait to merge this until the issue is solved @CartoDB/systems .

**Comment by lbosque** [comment]
@Kartones is this url going to be cached on varnish? I need to know that in order to do a change or another in nginx

**Comment by Kartones** [comment]
Yes, it is, I'm precisely working on that right now.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4610/))

**Pull request merged**